nixos/modules/services/display-managers/sddm.nix

```diff
@@ -234,20 +247,23 @@ in
 
     assertions = [
       {
-        assertion = xcfg.enable;
+        assertion = xcfg.enable || cfg.wayland.enable;
         message = ''
-          SDDM requires services.xserver.enable to be true
+          SDDM requires either services.xserver.enable or services.displayManager.sddm.wayland.enable to be true
         '';
       }
       {
-        assertion = dmcfg.autoLogin.enable -> autoLoginSessionName != null;
+        assertion = config.services.displayManager.autoLogin.enable -> autoLoginSessionName != null;
         message = ''
           SDDM auto-login requires that services.xserver.displayManager.defaultSession is set.
         '';
       }
     ];
 
-    services.xserver.displayManager.job.execCmd = "exec /run/current-system/sw/bin/sddm";
+    services.displayManager = {
+      enable = true;
```

Comment:
we could possibly set `systemd.defaultUnit = "graphical.target"` when sddm is enabled.

Greetd does something similar, and this is also set when having xserver enabled that is set my the xserver module.

That would then also mean, that users sddm users could migrate to a non xserver config by just removing the `services.xserver.enable`.

Reply:
Fixed in [`6ef35e8` (#291913)](https://github.com/NixOS/nixpkgs/pull/291913/commits/6ef35e8e4d4323c31db1c7f11e00d676cf6edb36)